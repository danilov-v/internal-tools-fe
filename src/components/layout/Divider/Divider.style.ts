import styled from 'styled-components';

type ContainerProps = {
  mb: number;
  mt: number;
};

export const Container = styled.div<ContainerProps>`
  height: 1px;
  background-color: ${({ theme }) => theme.lightGray};
  margin-bottom: ${({ mb }) => mb}px;
  margin-top: ${({ mt }) => mt}px;
  width: 100%;
`;
