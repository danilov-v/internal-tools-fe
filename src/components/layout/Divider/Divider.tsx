import React from 'react';

import * as S from './Divider.style';

const Divider: React.FC<DividerProps> = ({ mb = 8, mt = 8 }) => (
  <S.Container mb={mb} mt={mt} />
);

type DividerProps = {
  mb?: number;
  mt?: number;
};

export { Divider };
