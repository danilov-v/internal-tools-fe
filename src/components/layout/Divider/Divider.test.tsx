import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import 'jest-styled-components';

import { Divider } from './Divider';

const getComponent = (props = {}): RenderResult => {
  const parsedProps = {
    ...props,
  };

  return render(<Divider {...parsedProps} />);
};

describe('Divider component', () => {
  it('matches snapshot with only required props', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });

  it('matches snapshot with additional props', () => {
    const component = getComponent({
      children: 'children',
      mb: 100,
      mt: 20,
    });

    expect(component.baseElement).toMatchSnapshot();
  });
});
