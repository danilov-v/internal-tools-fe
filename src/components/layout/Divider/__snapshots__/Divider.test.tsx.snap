// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component matches snapshot with additional props 1`] = `
<body>
  .c0 {
  height: 1px;
  margin-bottom: 100px;
  margin-top: 20px;
  width: 100%;
}

<div>
    <div
      class="c0"
    />
  </div>
</body>
`;

exports[`Divider component matches snapshot with only required props 1`] = `
.c0 {
  height: 1px;
  margin-bottom: 8px;
  margin-top: 8px;
  width: 100%;
}

<body>
  <div>
    <div
      class="c0"
    />
  </div>
</body>
`;
